File: low-level-design/design-patterns/python/structural/facade-pattern/gcp_manager.py
from resources import BlockStorageManager,NetworkManager,VMManager

class GCPBlockStorageManager(BlockStorageManager):
    def create(self):
        print("GCP Block Storage Created")
    def delete(self):
        print("GCP Block Storage Deleted")

class GCPNetworkManager(NetworkManager):
    def create(self):
        print("GCP Network Created")
    def delete(self):
        print("GCP Network Deleted")

class GCPVMManager(VMManager):
    def create(self):
        print("GCP VM Created")
    def delete(self):
        print("GCP VM Deleted")
    def start(self):
        print("GCP VM Started")
    def stop(self):
        print("GCP VM Stopped")

File: low-level-design/design-patterns/golang/structural/facade-pattern/facade/gcp_manager.go
package facade

import "fmt"

func NewGCPBlockStorageManager() GCPBlockStorageManager {
	return GCPBlockStorageManager{}
}

type GCPBlockStorageManager struct {
}

func (m *GCPBlockStorageManager) Create()error {
	// Implementation for creating a GCP block storage
	fmt.Println("creating GCP block storage")
	return nil 
}
func (m *GCPBlockStorageManager) Delete()error{
	// Implementation for deleting a GCP block storage
	fmt.Println("deleting GCP block storage")
	return nil 
}

func NewGCPNetworkManager() GCPNetworkManager {
	return GCPNetworkManager{}
}

type GCPNetworkManager struct {
}

func (m *GCPNetworkManager) Create() error {
	// Implementation for creating a GCP network
	fmt.Println("creating GCP network")
	return nil
}
func (m *GCPNetworkManager) Delete() error {
	// Implementation for deleting a GCP network
	fmt.Println("deleting GCP network")
	return nil
}

func NewGCPVMManager() GCPVMManager {
	return GCPVMManager{}
}

type GCPVMManager struct {
}

func (m *GCPVMManager) Create() error {
	// Implementation for creating a GCP VM
	fmt.Println("creating GCP VM")
	return m.Start()
}

func (m *GCPVMManager) Delete() error {
	// Implementation for deleting a GCP VM
	fmt.Println("deleting GCP VM")
	return nil
}
func (m *GCPVMManager) Start() error {

	fmt.Println("starting GCP VM")
	return nil
}

func (m *GCPVMManager) Stop() error {
	// Implementation for stopping a GCP VM
	fmt.Println("stopping GCP VM")
	return nil
}
